src/Foundatio/Storage/IFileStorage.cs

```diff
@@ -18,7 +19,61 @@ public interface IFileStorage : IHaveSerializer, IDisposable {
         Task<bool> CopyFileAsync(string path, string targetPath, CancellationToken cancellationToken = default);
         Task<bool> DeleteFileAsync(string path, CancellationToken cancellationToken = default);
         Task DeleteFilesAsync(string searchPattern = null, CancellationToken cancellation = default);
-        Task<IEnumerable<FileSpec>> GetFileListAsync(string searchPattern = null, int? limit = null, int? skip = null, CancellationToken cancellationToken = default);
+        Task<FileListResult> GetFileListAsync(string searchPattern = null, int? limit = null, CancellationToken cancellationToken = default);
+    }
+
+    public interface IHasNextPageFunc {
+        Func<Task<NextPageResult>> NextPageFunc { get; set; }
+    }
+
+    public class NextPageResult {
+        public bool Success { get; set; }
+        public bool HasMore { get; set; }
+        public IReadOnlyCollection<FileSpec> Files { get; set; }
+        public Func<Task<NextPageResult>> NextPageFunc { get; set; }
+    }
+
+    public class FileListResult : IHasNextPageFunc {
+        private static IReadOnlyCollection<FileSpec> _empty = new ReadOnlyCollection<FileSpec>(new FileSpec[0]);
+        public static FileListResult Empty = new FileListResult(_empty);
+
+        public FileListResult(IReadOnlyCollection<FileSpec> files) {
+            Files = files;
+            HasMore = false;
+            ((IHasNextPageFunc)this).NextPageFunc = null;
```

Comment:
These should be defaulted values.

Reply:
Huh?

Reply:
Any reason for setting a value which is already the default value of both HasMore (false) and NextPageFunc (null)